paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -1305,6 +1321,76 @@ def diagnose_why_instance_is_stuck(
             print(f"    {line}")
     print("")
 
+    if should_ping_for_unhealthy_pods:
+        maybe_ping_for_unhealthy_pods(
+            service, instance, cluster, git_sha, status, notify_fn
+        )
+
+
+already_pinged = False
+
+
+def maybe_ping_for_unhealthy_pods(
+    service: str,
+    instance: str,
+    cluster: str,
+    git_sha: str,
+    status: InstanceStatusKubernetesV2,
+    notify_fn: Optional[Callable[[str], None]] = None,
+) -> None:
+    if notify_fn and not already_pinged:
+        # there can be multiple current versions, e.g. if someone changes yelpsoa-configs during a bounce.
+        current_versions = [
+            v for v in status.kubernetes_v2.versions if v.git_sha == git_sha
+        ]
+        pingable_pods = [
+            pod
+            for version in current_versions
+            for pod in version.pods
+            if should_ping_for_pod(pod)
+        ]
+        if pingable_pods:
+            ping_for_pods(service, instance, cluster, pingable_pods, notify_fn)
+
+
+def should_ping_for_pod(pod: KubernetesPodV2) -> bool:
+    return recent_container_restart(get_main_container(pod))
+
+
+def ping_for_pods(
+    service: str,
+    instance: str,
+    cluster: str,
+    pods: List[KubernetesPodV2],
+    notify_fn: Callable[[str], None],
+) -> None:
+    global already_pinged
+    already_pinged = True
+
+    pods_by_reason: Dict[str, List[KubernetesPodV2]] = {}
+    for pod in pods:
+        pods_by_reason.setdefault(get_main_container(pod).reason, []).append(pod)
+
+    for reason, pods_with_reason in pods_by_reason.items():
+        explanation = {
+            "Error": "crashed on startup",
+            "OOMKilled": "run out of memory",
+            "CrashLoopBackOff": "crashed on startup several times, and Kubernetes is backing off restarting them",
+        }.get(reason, f"restarted ({reason})")
+
+        tip = {
+            "Error": (
+                "This may indicate a bug in your code, a misconfiguration in yelpsoa-configs, or missing srv-configs. "
+                f"Take a look at the output of your unhealthy pods with `paasta status -s {service} -i {instance} -c {cluster} -vv` (more -v for more output.)"
+            ),
+            "CrashLoopBackOff": "This may indicate a bug in your code, a misconfiguration in yelpsoa-configs, or missing srv-configs.",
+            "OOMKilled": "This probably means your new version of code requires more memory than the old version. You may want to increase memory in yelpsoa-configs or roll back.",
+        }.get(reason, "")
```

Comment:
We might want to suggest asking for help for paasta owners, since we're otherwise not providing debugging help.